raster-foundry/raster-foundry
# Refactor Docker Compose services and remove duplicate Backsplash files

## Overview

* Remove `app-backend/Dockerfile.backsplash`
* Remove `app-backend/backsplash` directory
* Use [azavea/docker-openjdk-gdal](https://github.com/azavea/docker-openjdk-gdal) base image for Backsplash in place of daunnc/openjdk-gdal base image
* Remove duplicate service definitions from `docker-compose.java.yml`
* Refactor `scripts/console`
* Refactor `scripts/server`
* Refactor `scripts/cibuild`
* Refactor `scripts/test`
* Refactor `scripts/update`

Resolves #4464 

## Notes

Running `./scripts/server` will no longer target `docker-compose.java.yml` by default. 

This PR introduces the `--java` flag to `./scripts/server`, and makes it so `docker-compose.java.yml` acts as a command override to the `api-server` and `backsplash` service definitions in `docker-compose.yml`.

## Testing Instructions

Testing these changes proved challenging because I found it difficult to successfully bring up a development environment for the project, even on `develop`. 

I think I have been able to replicate these steps:

First, make sure your workspace is clean, there is no VM created, and then run `setup`:

```bash
$ vagrant destroy -f
$ git clean -dxff
$ git reset --hard
$ ./scripts/setup
```

Next, load in the development database:

```bash
$ vagrant ssh
vagrant@vagrant:/opt/raster-foundry$ sudo ./scripts/load_development_data --download
```

Next, apply migrations:

```bash
vagrant@vagrant:/opt/raster-foundry$ docker-compose run --rm api-server "mg update"
vagrant@vagrant:/opt/raster-foundry$ docker-compose run --rm api-server "mg apply"
```

Finally, run `server` and make sure the web application works:

```bash
vagrant@vagrant:/opt/raster-foundry$ ./scripts/server
```

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/1774125/52871522-3d5d1f80-3118-11e9-9327-37a561a85c1f.png">
